skillserver/skillserver.go

```diff
@@ -58,12 +59,38 @@ func SetRootPrefix(prefix string) {
 	RootPrefix = prefix
 }
 
+// RunSSL takes in a map of application, server port, certificate and key file names, and
+// tries to start a TLS server which alexa can directly pass commands to.
+// It returns back an error if the server couldn't be started. Or else the method blocks
+// at ListenAndServeTLS line
+// IF the server starts succcessfully and there are connection errors afterwards, they are
+// logged to the stdout and no error is returned
+// For generating a testing cert and key, read the following:
+// https://developer.amazon.com/docs/custom-skills/configure-web-service-self-signed-certificate.html
 func RunSSL(apps map[string]interface{}, port, cert, key string) error {
-	router := mux.NewRouter()
-	Init(apps, router)
-
-	err := http.ListenAndServeTLS(port, cert, key, router)
-	return err
+  router := mux.NewRouter()
+  Init(apps, router)
+
+  // This is very limited TLS configuration which is required to connect alexa to our webservice.
+  // The curve preferences are used by ECDSA/ECDHE algorithms for figuring out the mathing algorithm
+  // from alexa side starting from the strongest to the weakest
+  cfg := &tls.Config{
+    MinVersion:               tls.VersionTLS12,
+    CurvePreferences:         []tls.CurveID{tls.CurveP521, tls.CurveP384, tls.CurveP256},
+    PreferServerCipherSuites: true,
+    CipherSuites: []uint16{
+      //If the connection throws errors related to crypt algorithm mismatch between server and client,
+      //this line must be replaced by constants present in crypt/tls package for the value that works
+      tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
+    },
+  }
+  srv := &http.Server{
+    Addr:         ":" + port,
```

Comment:
the original code didn't prefix with a ":". Do we really want to do that here? Maybe something like:

```go
if !strings.HasPrefix(port, ":") {
    port = ":" + port
}
```

would be better?

Reply:
I was actually trying to make it more consistent with the Run method above so that people don't have to put their brains while switching from non ssl to ssl and vice versa. Would it be a good idea to return the error from Run instead of firing n.Run() which actually does log.Fatal at the end of execution, or change our method to do a log.Fatal instead of returning err?

Reply:
As far as the port is concerned, its actually a port value, and we listen to all the interfaces in the machine, so I'd like to stick to the current port implementation.

Reply:
I agree it breaks backward compatibility, but it seems the function actually wasn't working. I hope people aren't using it, hence it could give us some flexibility around changing the whole method
I like the idea of MustRunSSL. Should I put that in as well, or just leave it as is, it has already been merged to the main line now..

Reply:
I think it's fine for now. I would just try to keep it in mind going forward. Backwards compatibility is important.

Reply:
Btw, thanks for the contribution!